lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystemE2E.cs

```diff
@@ -228,6 +229,21 @@ public void TestCopyFromLocalE2E()
             container.DeleteIfExistsAsync().Wait();
         }
 
+        [Fact(Skip = SkipMessage)]
```

Comment:
I know this is following the existing pattern -- but instead of having to remove all the skips and then set credentials, can we change this to dynamically skip the test if the credentials are not set? It looks like xunit does support a SkipTestException: https://github.com/xunit/samples.xunit/blob/master/DynamicSkipExample/SkipTestException.cs

Not sure if this is supported in our version of xunit or not.

Reply:
I would also put the check in a single function and call it before the test.

Reply:
OK sounds good, it maybe worth making that a bit more clear above ( there is a comment off to the side that says use null to run tests. It maybe more clear to have another like under SkipMessage with SkipMessage = null commented out and a comment above to uncomment to run skipped tests.

---
In reply to: [196930627](https://github.com/apache/reef/pull/1473#discussion_r196930627) [](ancestors = 196930627,195885915)

Reply:
Keeping this as it is as discussed offline

Reply:
Can you please elaborate? We might need to refer to this PR in a future when we have all forgotten what was discussed offline :)

Reply:
Scott said it is fine to have the current line. 
 private const string SkipMessage = "Fill in credentials before running test"; // Use null to run tests

Reply:
I did -- so setting SkipMessage=null has the same effect as the SkipTestException since our version of XUnit does not support it. But I think the comment of "// Use null to run tests" could be a bit more explicit. This is more what I was thinking:
```
//Uncomment SkipMessage=null to run tests
SkipMessage="....";
//SkipMessage=null
```

Reply:
Changed as suggested